### ISSUE TITLE
BE OOM when use rpad function

### ISSUE THREAD

**Issue by wangruin** [StarRocks/starrocks#2958]
### Steps to reproduce the behavior (Required)
1. use tpch 100g data
2. CREATE TABLE `tpch_100g_customer` (
  `C_CUSTKEY` int(11) NOT NULL COMMENT "",
  `C_NAME` varchar(25) NOT NULL COMMENT "",
  `C_ADDRESS` varchar(40) NOT NULL COMMENT "",
  `C_NATIONKEY` int(11) NOT NULL COMMENT "",
  `C_PHONE` char(15) NOT NULL COMMENT "",
  `C_ACCTBAL` decimal64(15, 2) NOT NULL COMMENT "",
  `C_MKTSEGMENT` char(10) NOT NULL COMMENT "",
  `C_COMMENT` varchar(117) NOT NULL COMMENT ""
) ENGINE=OLAP
DUPLICATE KEY(`C_CUSTKEY`)
COMMENT "OLAP"
DISTRIBUTED BY HASH(`C_CUSTKEY`) BUCKETS 12
PROPERTIES (
"replication_num" = "1",
"in_memory" = "false",
"storage_format" = "DEFAULT"
);
2. select   * from   sqlsmith.tpch_100g_customer as ref_0 where      cast(       rpad(         cast(ref_0.C_MKTSEGMENT as VARCHAR),         cast(ref_0.C_CUSTKEY as INT),         cast(ref_0.C_PHONE as VARCHAR)       ) as VARCHAR     ) > ref_0.C_MKTSEGMENT;

### Expected behavior (Required)
Query returns correctly

### Real behavior (Required)
```
start time: Thu Jan 20 11:17:57 CST 2022
tcmalloc: large alloc 2103001088 bytes == 0x203636000 @  0x4d51259 0x4ecec1c 0x1cc82e8 0x4e1f4c5 0x2f6b47a 0x2f6b80e 0x2f6bb4b 0x2cc56a6 0x28855bc 0x27de9b4 0x20c9531 0
x249e4bb 0x21ed805 0x1c1a820 0x3411fb7
tcmalloc: large alloc 2114822144 bytes == 0x281068000 @  0x4d51259 0x4ecec1c 0x1cc82e8 0x4e1f4c5 0x2f6b47a 0x2f6b80e 0x2f6bb4b 0x2cc56a6 0x28855bc 0x27de9b4 0x20c9531 0
x249e4bb 0x21ed805 0x1c1a820 0x3411fb7
tcmalloc: large alloc 2114682880 bytes == 0x281068000 @  0x4d51259 0x4ecec1c 0x1cc82e8 0x4e1f4c5 0x2f6b47a 0x2f6b80e 0x2f6bb4b 0x2cc56a6 0x28855bc 0x27de9b4 0x20c9531 0
x249e4bb 0x21ed805 0x1c1a820 0x3411fb7
tcmalloc: large alloc 2108612608 bytes == 0x2ff120000 @  0x4d51259 0x4ecec1c 0x1cc82e8 0x4e1f4c5 0x2f6b47a 0x2f6b80e 0x2f6bb4b 0x2cc56a6 0x28855bc 0x27de9b4 0x20c9531 0
x249e4bb 0x21ed805 0x1c1a820 0x3411fb7
tcmalloc: large alloc 2099937280 bytes == 0x203636000 @  0x4d51259 0x4ecec1c 0x1cc82e8 0x4e1f4c5 0x2f6b47a 0x2f6b80e 0x2f6bb4b 0x2cc56a6 0x28855bc 0x27de9b4 0x20c9531 0
x249e4bb 0x21ed805 0x1c1a820 0x3411fb7
tcmalloc: large alloc 2091016192 bytes == 0x281068000 @  0x4d51259 0x4ecec1c 0x1cc82e8 0x4e1f4c5 0x2f6b47a 0x2f6b80e 0x2f6bb4b 0x2cc56a6 0x28855bc 0x27de9b4 0x20c9531 0
x249e4bb 0x21ed805 0x1c1a820 0x3411fb7
tcmalloc: large alloc 4070023168 bytes == 0x3f92ce000 @  0x4d51259 0x4ecec1c 0x1cc82e8 0x4e1f4c5 0x2f6b47a 0x2f6b80e 0x2f6bb4b 0x2cc56a6 0x28855bc 0x27de9b4 0x20c9531 0x249e4bb 0x21ed805 0x1c1a820 0x3411fb7
tcmalloc: large alloc 4168253440 bytes == 0x4ebc48000 @  0x4d51259 0x4ecec1c 0x1cc82e8 0x4e1f4c5 0x2f6b47a 0x2f6b80e 0x2f6bb4b 0x2cc56a6 0x28855bc 0x27de9b4 0x20c9531 0x249e4bb 0x21ed805 0x1c1a820 0x3411fb7
tcmalloc: large alloc 4069720064 bytes == 0x5e4b70000 @  0x4d51259 0x4ecec1c 0x1cc82e8 0x4e1f4c5 0x2f6b47a 0x2f6b80e 0x2f6bb4b 0x2cc56a6 0x28855bc 0x27de9b4 0x20c9531 0x249e4bb 0x21ed805 0x1c1a820 0x3411fb7
tcmalloc: large alloc 4061634560 bytes == 0x5e4b70000 @  0x4d51259 0x4ecec1c 0x1cc82e8 0x4e1f4c5 0x2f6b47a 0x2f6b80e 0x2f6bb4b 0x2cc56a6 0x28855bc 0x27de9b4 0x20c9531 0x249e4bb 0x21ed805 0x1c1a820 0x3411fb7
tcmalloc: large alloc 4158840832 bytes == 0x3f92ce000 @  0x4d51259 0x4ecec1c 0x1cc82e8 0x4e1f4c5 0x2f6b47a 0x2f6b80e 0x2f6bb4b 0x2cc56a6 0x28855bc 0x27de9b4 0x20c9531 0x249e4bb 0x21ed805 0x1c1a820 0x3411fb7
tcmalloc: large alloc 4034510848 bytes == 0x4f10fc000 @  0x4d51259 0x4ecec1c 0x1cc82e8 0x4e1f4c5 0x2f6b47a 0x2f6b80e 0x2f6bb4b 0x2cc56a6 0x28855bc 0x27de9b4 0x20c9531 0x249e4bb 0x21ed805 0x1c1a820 0x3411fb7
tcmalloc: large alloc 7696465920 bytes == 0x7ccf2c000 @  0x4d51259 0x4ecec1c 0x1cc82e8 0x4e1f4c5 0x2f6b47a 0x2f6b80e 0x2f6bb4b 0x2cc56a6 0x28855bc 0x27de9b4 0x20c9531 0x249e4bb 0x21ed805 0x1c1a820 0x3411fb7
tcmalloc: large alloc 7680188416 bytes == 0x998318000 @  0x4d51259 0x4ecec1c 0x1cc82e8 0x4e1f4c5 0x2f6b47a 0x2f6b80e 0x2f6bb4b 0x2cc56a6 0x28855bc 0x27de9b4 0x20c9531 0x249e4bb 0x21ed805 0x1c1a820 0x3411fb7
terminate called after throwing an instance of 'std::bad_alloc'
  what():  std::bad_alloc
*** Aborted at 1642648735 (unix time) try "date -d @1642648735" if you are using GNU date ***
PC: @     0x7f07e4b3e337 __GI_raise
*** SIGABRT (@0x3e80001e4fb) received by PID 124155 (TID 0x7f07c7bc8700) from PID 124155; stack trace: ***
    @          0x346a1c2 google::(anonymous namespace)::FailureSignalHandler()
    @     0x7f07e58095f0 (unknown)
    @     0x7f07e4b3e337 __GI_raise
    @     0x7f07e4b3fa28 __GI_abort
    @          0x15c3023 _ZN9__gnu_cxx27__verbose_terminate_handlerEv.cold
    @          0x4e1ef66 __cxxabiv1::__terminate()
    @          0x4e1efd1 std::terminate()
    @          0x4e1f125 __cxa_throw
    @          0x15c2f2a _Znwm.cold
    @          0x1732750 std::vector<>::_M_default_append()
    @          0x2f6b47a starrocks::vectorized::pad_not_const<>()
    @          0x2f6b80e starrocks::vectorized::pad_not_const_check_ascii<>()
    @          0x2f6bb4b starrocks::vectorized::StringFunctions::rpad()
    @          0x2cc56a6 starrocks::vectorized::VectorizedFunctionCallExpr::evaluate()
    @          0x28855bc starrocks::vectorized::VectorizedBinaryPredicate<>::evaluate()
    @          0x27de9b4 starrocks::ExprContext::evaluate()
    @          0x20c9531 starrocks::ExecNode::eval_conjuncts()
    @          0x249e4bb starrocks::vectorized::TabletScanner::get_chunk()
    @          0x21ed805 starrocks::vectorized::OlapScanNode::_scanner_thread()
    @          0x1c1a820 starrocks::PriorityThreadPool::work_thread()
    @          0x3411fb7 thread_proxy
    @     0x7f07e5801e65 start_thread
    @     0x7f07e4c0688d __clone
    @                0x0 (unknown)
(END)
```

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
branch-2.1 release 4995053

**Comment by wangruin** [comment]
This SQL will also cause the same crash, see if it is the same problem
```
select   * from   sqlsmith.tpch_100g_part as ref_0 where   (     ltrim(cast(ref_0.P_NAME as VARCHAR)) = lpad(       cast(ref_0.P_NAME as VARCHAR),       cast(ref_0.P_PARTKEY as INT),       cast(ref_0.P_MFGR as VARCHAR)     )   );
```

**Comment by stdpain** [comment]
fixed by #4368
